demos/coreMQTT/mqtt_demo_keep_alive.c

```diff
@@ -0,0 +1,1172 @@
+#include <stdlib.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+#include "timers.h"
+
+/* MQTT library includes. */
+#include "core_mqtt.h"
+
+/* Retry utilities include. */
+#include "retry_utils.h"
+
+/* Transport interface include. */
+#include "transport_secure_sockets.h"
+
+/* Client credential include for the default MQTT broker endpoint. */
+#include "aws_clientcredential.h"
+
+/*-----------------------------------------------------------*/
+
+/* Compile time error for undefined configs. */
+
+#ifndef democonfigMQTT_BROKER_ENDPOINT
+    #define democonfigMQTT_BROKER_ENDPOINT    clientcredentialMQTT_BROKER_ENDPOINT
+#endif
+
+/**
+ * @brief The MQTT client identifier used in this example.  Each client identifier
+ * must be unique so edit as required to ensure no two clients connecting to the
+ * same broker use the same client identifier.
+ */
+#ifndef democonfigCLIENT_IDENTIFIER
+    #define democonfigCLIENT_IDENTIFIER    clientcredentialIOT_THING_NAME
+#endif
+
+/**
+ * @brief The port to use for the demo.
+ */
+#ifndef democonfigMQTT_BROKER_PORT
+    #define democonfigMQTT_BROKER_PORT    clientcredentialMQTT_BROKER_PORT
+#endif
+
+/**
+ * @brief The maximum number of times to run the subscribe publish loop in this
+ * demo.
+ */
+#ifndef democonfigMQTT_MAX_DEMO_COUNT
+    #define democonfigMQTT_MAX_DEMO_COUNT    ( 3 )
+#endif
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Timeout for receiving CONNACK packet in milliseconds.
+ */
+#define mqttexampleCONNACK_RECV_TIMEOUT_MS           ( 1000U )
+
+/**
+ * @brief The topic to subscribe and publish to in the example.
+ *
+ * The topic name starts with the client identifier to ensure that each demo
+ * interacts with a unique topic name.
+ */
+#define mqttexampleTOPIC                             democonfigCLIENT_IDENTIFIER "/example/topic"
+
+/**
+ * @brief The number of topic filters to subscribe.
+ */
+#define mqttexampleTOPIC_COUNT                       ( 1 )
+
+/**
+ * @brief The MQTT message published in this example.
+ */
+#define mqttexampleMESSAGE                           "Hello World!"
+
+/**
+ * @brief Dimensions a file scope buffer currently used to send and receive MQTT data
+ * from a socket.
+ */
+#define mqttexampleSHARED_BUFFER_SIZE                ( 500U )
+
+/**
+ * @brief Time to wait between each cycle of the demo implemented by
+ * RunCoreMqttPlaintextDemo().
+ */
+#define mqttexampleDELAY_BETWEEN_DEMO_ITERATIONS     ( pdMS_TO_TICKS( 5000U ) )
+
+/**
+ * @brief Timeout for MQTT_ReceiveLoop in milliseconds.
+ */
+#define mqttexampleRECEIVE_LOOP_TIMEOUT_MS           ( 500U )
+
+/**
+ * @brief Keep alive time reported to the broker while establishing an MQTT connection.
+ *
+ * It is the responsibility of the Client to ensure that the interval between
+ * Control Packets being sent does not exceed the this Keep Alive value. In the
+ * absence of sending any other Control Packets, the Client MUST send a
```

Comment:
```suggestion
 * absence of sending any other control packets, the client MUST send a
```

Reply:
sure i will get rid of all weird capitalizations.